torch/nn/modules/module.py

```diff
@@ -382,6 +382,137 @@ def add_module(self, name: str, module: Optional['Module']) -> None:
+                ``target`` if it exists, otherwise ``None``. ``None``
+                will also be returned if the target string resolves
+                to something that is not an ``nn.Module``.
+        """
+        atoms: List[str] = target.split(".")
+        mod: torch.nn.Module = self
+
+        for item in atoms:
+
+            if not hasattr(mod, item):
+                return None
+
+            mod = getattr(mod, item)
+
+            if not isinstance(mod, torch.nn.Module):
+                return None
+
+        return mod
+
+    def get_parameter(self, target: str) -> Optional["Parameter"]:
+        """
+        Returns the parameter given by ``target`` if it exists,
+        otherwise returns ``None``.
+
+        See the docstring for ``get_submodule`` for a more detailed
+        explanation of this method's functionality as well as how to
+        correctly specify ``target``.
+
+        Args:
+            target: The fully-qualified string name of the Parameter
+                to look for. (See ``get_submodule`` for how to specify a
+                fully-qualified string.)
+
+        Returns:
+            Optional[torch.nn.Parameter]: The Parameter referenced by
+                ``target`` if it exists, otherwise ``None``. ``None``
+                will also be returned if the target string resolves
+                to something that is not an ``nn.Parameter``.
+        """
+        module_path, _, param_name = target.rpartition(".")
+
+        mod: Optional[torch.nn.Module] = self.get_submodule(module_path)
+
+        if not mod:
+            return None
+
+        if not hasattr(mod, param_name):
+            return None
+
+        param: torch.nn.Parameter = getattr(mod, param_name)
+
+        if not isinstance(param, torch.nn.Parameter):
+            return None
+
+        return param
+
+    def get_buffer(self, target: str) -> Optional["Tensor"]:
+        """
+        Returns the buffer given by ``target`` if it exists,
+        otherwise returns ``None``.
+
+        See the docstring for ``get_submodule`` for a more detailed
+        explanation of this method's functionality as well as how to
+        correctly specify ``target``.
+
+        Args:
+            target: The fully-qualified string name of the buffer
+                to look for. (See ``get_submodule`` for how to specify a
+                fully-qualified string.)
+
+        Returns:
+            Optional[torch.Tensor]: The buffer referenced by
+                ``target`` if it exists, otherwise ``None``. ``None``
+                will also be returned if the target string resolves
+                to something that is not a buffer.
+        """
+        module_path, _, buffer_name = target.rpartition(".")
+
+        mod: Optional[torch.nn.Module] = self.get_submodule(module_path)
+
+        if not mod:
+            return None
+
+        for name, buffer in mod.named_buffers():
+            if name == buffer_name:
+                return buffer
```

Comment:
This is O(N). Can we instead directly fetch the attribute `getattr(mod, buffer_name)` and assert that `buffer_name` is in `self._buffers`?